core/common/src/main/java/alluxio/collections/FieldIndex.java

```diff
@@ -22,13 +22,14 @@
  *
  * @param <T> type of objects in this {@link IndexedSet}
  */
-interface FieldIndex<T> extends Iterable<T> {
+public interface FieldIndex<T> extends Iterable<T> {
   /**
    * Adds an object o to the index.
    *
    * @param o the object to add to the index
+   * @return true if success, otherwise false
```

Comment:
@gpang thank you for the comment, finally I changed this function back as returning void. Because if we a remove may happen before we return true thus make things inconsistent.